webapp/dojo_news/src/App.spec.js

```diff
@@ -0,0 +1,221 @@
+    it('list order is reversed upon Reverse-Button click from initial state _descending_', async () => {
+        const wrapper = shallowMount(App, {
+            propsData: {
+                initialNewsListItems: [
+                    {id: 0, title: "macOS", votes: 4},
+                    {id: 1, title: "Linux", votes: 8},
+                    {id: 2, title: "centOS", votes: 3},
+                    {id: 3, title: "Windows", votes: 0}
+                ],
+                initialDescendingOrder: true
+            }
+        });
+
+        let expectedIdList = [1, 0, 2, 3];
+        // or check HTML
+        let actualIdList = wrapper.vm.sortedNewsListItems.map(x => x.id);
+        expect(expectedIdList).toEqual(actualIdList);
+
+        await wrapper.find("#reverse-order-button").trigger("click");
+        expectedIdList = expectedIdList.reverse();
+        actualIdList = wrapper.vm.sortedNewsListItems.map(x => x.id);
+        expect(expectedIdList).toEqual(actualIdList);
+    });
+
+    it('list order is reversed upon Reverse-Button click from initial state _ascending_', async () => {
+        const wrapper = shallowMount(App, {
+            propsData: {
+                initialNewsListItems: [
+                    {id: 0, title: "macOS", votes: 4},
+                    {id: 1, title: "Linux", votes: 8},
+                    {id: 2, title: "centOS", votes: 3},
+                    {id: 3, title: "Windows", votes: 0}
+                ],
+                initialDescendingOrder: false
+            }
+        });
+
+        let expectedIdList = [3, 2, 0, 1];
+        let actualIdList = wrapper.vm.sortedNewsListItems.map(x => x.id);
+        expect(expectedIdList).toEqual(actualIdList);
+
+        await wrapper.find("#reverse-order-button").trigger("click");
+        expectedIdList = expectedIdList.reverse();
+        actualIdList = wrapper.vm.sortedNewsListItems.map(x => x.id);
+        expect(expectedIdList).toEqual(actualIdList);
+    });
+
+    it('displays the Reverse-Order-Button on an initially non empty list of NewsItems', () => {
+        const wrapper = shallowMount(App, {
+            propsData: {
+                initialNewsListItems: [
+                    {id: 0, title: "macOS", votes: 0},
+                    {id: 1, title: "Linux", votes: 0},
+                    {id: 2, title: "Windows", votes: 0}
+                ]
+            }
+        });
+
+        expect(wrapper.text()).toContain("Reverse Order");
+    });
+
+    it('does not display the Reverse-Order-Button on an initially empty list of NewsItems', () => {
+        const wrapperWithEmptyList = shallowMount(App, {
+            propsData: {
+                initialNewsListItems: []
+            }
+        });
+
+        expect(wrapperWithEmptyList.find("#reverse-order-button").element.hasAttribute("disabled")).toBeTruthy();
+        expect(wrapperWithEmptyList.find("#reverse-order-button").element.getAttribute("disabled")).toEqual("disabled");
```

Comment:
```suggestion
        expect(wrapperWithEmptyList.find("#reverse-order-button").attributes("disabled")).toEqual("disabled");
```
See: https://vue-test-utils.vuejs.org/api/wrapper/#attributes

Reply:
Fixed, thanks!